app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,36 @@
+class CategoriesController < ApplicationController
+before_action :require_login, only: [:new, :create, :edit, :update]
+before_action :navbar_categories, only: [:index, :show]
+
+  def index
+    @cat = Category.all
+  end
+  def show
+    @category = Category.find(params[:id])
+  end
+  def new
+    @category = Category.new
+    @action = "create"
+  end
+  def create
+    @category = Category.new(category_params)
+    @category.user_id = session[:user_id]
+    if @category.save
+      redirect_to user_products_dash_path(@category.user_id)
+    else
+      render :new
+    end
+  end
+  def edit
+
+  end
```

Comment:
Since there's nothing defined in these standard methods, you could actually leave them off altogether, they'll still work! BUT it's nice to be explicit and do what Lindsey suggested.